kornia/geometry/bbox_v2.py

```diff
@@ -0,0 +1,585 @@
+
+    # Work with batch as kornia.transform_points only supports a batch of points.
+    boxes_per_batch, n_points_per_box, coordinates_dimension = boxes.shape[-3:]
+    points = boxes.view(-1, n_points_per_box * boxes_per_batch, coordinates_dimension)
+    M = M if M.ndim == 3 else M.unsqueeze(0)
+
+    if points.shape[0] != M.shape[0]:
+        raise ValueError(
+            f"Batch size mismatch. Got {points.shape[0]} for hexahedron, {M.shape[0]} for the transformation matrix."
+        )
+
+    transformed_boxes: torch.Tensor = kornia.transform_points(M, points)
+    transformed_boxes = transformed_boxes.view_as(boxes)
+    return transformed_boxes
+
+
+def _boxes_to_polygons(
+    xmin: torch.Tensor, ymin: torch.Tensor, width: torch.Tensor, height: torch.Tensor
+) -> torch.Tensor:
+    if not xmin.ndim == ymin.ndim == width.ndim == height.ndim == 2:
+        raise ValueError("We expect to create a batch of hexahedron in vertices format (B, N, 4, 2)")
+
+    # Create (B,N,4,2) with all points in top left position of the bounding box
+    polygons = torch.zeros((xmin.shape[0], xmin.shape[1], 4, 2), device=xmin.device, dtype=xmin.dtype)
+    polygons[..., 0] = xmin.unsqueeze(-1)
+    polygons[..., 1] = ymin.unsqueeze(-1)
+    # Shift top-right, bottom-right, bottom-left points to the right coordinates
+    polygons[..., 1, 0] += width  # Top right
+    polygons[..., 2, 0] += width  # Bottom right
+    polygons[..., 2, 1] += height  # Bottom right
+    polygons[..., 3, 1] += height  # Bottom left
+    return polygons
+
+
+def _boxes3d_to_polygons3d(
+    xmin: torch.Tensor,
+    ymin: torch.Tensor,
+    zmin: torch.Tensor,
+    width: torch.Tensor,
+    height: torch.Tensor,
+    depth: torch.Tensor,
+):
+    if not xmin.ndim == ymin.ndim == zmin.ndim == width.ndim == height.ndim == depth.ndim == 2:
+        raise ValueError("We expect to create a batch of 3D hexahedrons in vertices format (B, N, 8, 3)")
+
+    # Front
+    # Create (B,N,4,3) with all points in front top left position of the bounding box
+    front_vertices = torch.zeros((xmin.shape[0], xmin.shape[1], 4, 3), device=xmin.device, dtype=xmin.dtype)
+    front_vertices[..., 0] = xmin.unsqueeze(-1)
+    front_vertices[..., 1] = ymin.unsqueeze(-1)
+    front_vertices[..., 2] = zmin.unsqueeze(-1)
+    # Shift front-top-right, front-bottom-right, front-bottom-left points to the right coordinates
+    front_vertices[..., 1, 0] += width  # Top right
+    front_vertices[..., 2, 0] += width  # Bottom right
+    front_vertices[..., 2, 1] += height  # Bottom right
+    front_vertices[..., 3, 1] += height  # Bottom left
+
+    # Back
+    back_vertices = front_vertices.clone()
+    back_vertices[..., 2] += depth.unsqueeze(-1)
+
+    polygons3d = torch.cat([front_vertices, back_vertices], dim=-2)
+    return polygons3d
+
+
+@torch.jit.script
+class Boxes:
+    def __init__(self, quadrilaterals: torch.Tensor, raise_if_not_floating_point: bool = True):
```

Comment:
I think there is a bit conflicting between ```quadrilaterals ``` and ```Boxes```. Namely, boxes shall be rectangles, which is a special case of quadrilateral. If so, I think at least you should validate if the input is a correct rectangle than any random quadrilaterals.

Reply:
I'm not sure as users may want to leverage working with quadrilaterals. For example, some datasets like Open Image V6 provides the actual 4 extreme points made by the annotator. In this case, users may want to leverage this information to make boxes more accurate after transforms. 

I agree that the name is confusing. It would be better to call as `kornia_boxes` instead of `quadrilaterals`?. Then, in the class doc, we explain in which format kornia represents boxes. For example:
> Kornia 2D boxes format is defined as a floating data type tensor of shape ``Nx4x2`` or ``BxNx4x2`` where each box is a `quadrilateral <https://en.wikipedia.org/wiki/Quadrilateral>`_ defined by it's 4 vertices coordinates (A, B, C, D). Coordinates must be in ``x, y`` order. The height and width of a box is defined as ``width = xmax - xmin`` and ``height = ymax - ymin``. Examples of `quadrilaterals <https://en.wikipedia.org/wiki/Quadrilateral>`_ are rectangles, rhombus and trapezoids.

Reply:
I agree with Jian - so maybe we need both classes, one subclass the other?

Reply:
torchscript classes doesn't support inheritance. So, I would stick to one to avoid code duplication. 

I believe that the normal workflow will be to create `Boxes` class from `Boxes.from_tensor` as (xyxy) or (xywh) it's the standard format that users tend to work. I envision calling directly Boxes constructor as an advance usage. In that case, for me, it makes sense to give users freedom to represent boxes as they want to allow Openimage V6 use cases.